File: CPP/test/Test_Cheese.cpp
#include "../src/Cheese.cpp"
#include <gtest/gtest.h>

using namespace std;

TEST(CheeseTest, AllFunctionalityTest){
    Cheese testCheese(100);
    LinkedList<FarmProductType> CheeseRecipe = testCheese.getResep();
    ASSERT_EQ(MILK, CheeseRecipe.get(0));
    ASSERT_EQ(EGG, CheeseRecipe.get(1));
    ASSERT_EQ(MILK, testCheese.getResepwithIdx(0));
    ASSERT_EQ(EGG, testCheese.getResepwithIdx(1));
    ASSERT_EQ(100, testCheese.getPrice());
    ASSERT_EQ(CHEESE, testCheese.getObjectType());
    ASSERT_EQ(SIDEPRODUCT, testCheese.getProductType());
}

File: Java/src/test/java/engisubs/gameobject/product/sideproduct/TestCheese.java
package engisubs.gameobject.product.sideproduct;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import engisubs.gameobject.GameObject.FarmProductType;
import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;

public class TestCheese{
    private Cheese test;

    @Before
    public void setUpObject() {
        test = new Cheese(100);
    }

    @Test
    public void instantiationTest(){
        Assert.assertEquals(100, test.getPrice());
        Assert.assertEquals(GameObjectType.CHEESE, test.getGameObjectType());
        Assert.assertEquals(ProductType.SIDEPRODUCT, test.getProductType());
        Assert.assertEquals("Mozarella Cheese Italiano~~", test.getProductName());
    }
    @Test
    public void resepTest(){
        Cheese.InitResep();
        Assert.assertArrayEquals(new FarmProductType[]{FarmProductType.MILK, FarmProductType.EGG}, Cheese.getResep().toArray());
        Assert.assertEquals(null, Cheese.getResepwithIdx(3));
    }
}